disnake/threads.py

```diff
@@ -959,3 +1010,254 @@ def _from_data(self, data: ThreadMemberPayload):
+        payload["id"] = self.id
+        return payload
+
+    async def edit(
+        self,
+        *,
+        name: str = MISSING,
+        emoji: Optional[Union[str, Emoji, PartialEmoji]] = MISSING,
+        moderated: bool = MISSING,
+        reason: Optional[str] = None,
+    ) -> ThreadTag:
+        """|coro|
+
+        Edits the tag.
+
+        You must have the :attr:`~Permissions.manage_channels` permission to
+        do this.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The new tag name.
+        emoji: Optional[Union[:class:`str`, :class:`Emoji`, :class:`PartialEmoji`]]
+            The new tag emoji. Set to ``None`` to remove the emoji.
+        moderated: :class:`bool`
+            Whether only moderators can apply this tag to threads.
+        reason: Optional[:class:`str`]
+            The reason for editing the tag. Shows up on the audit log.
+
+        Raises
+        ------
+        InvalidData
+            Invalid channel/tag data was received from Discord.
+        Forbidden
+            You do not have permissions to edit the tag.
+        HTTPException
+            Editing the tag failed.
+
+        Returns
+        -------
+        :class:`ThreadTag`
+            The newly edited tag.
+        """
+        if isinstance(self._channel, Object):
+            raise TypeError("Cannot edit thread with no associated forum channel")
+
+        if self.id not in self._channel._available_tags:
+            # should only happen if tag was stored and tags were edited afterwards
+            raise ValueError("This tag is not part of the associated forum channel")
+
+        # we always have to patch the channel with the entire tag list,
+        # so serialize all tags and only replace the one matching `self.id`
+        new_tags = {tag_id: tag.to_dict() for tag_id, tag in self._channel._available_tags.items()}
+        # this always exists, see check above
+        new_tag_payload = new_tags[self.id]
+
+        if name is not MISSING:
+            new_tag_payload["name"] = name
+        if emoji is not MISSING:
+            emoji_name, emoji_id = PartialEmoji._to_name_id(emoji)
+            new_tag_payload["emoji_name"] = emoji_name
+            new_tag_payload["emoji_id"] = emoji_id
+        if moderated is not MISSING:
+            new_tag_payload["moderated"] = moderated
+
+        channel_data = cast(
+            "ForumChannelPayload",
+            await self._state.http.edit_channel(
+                self._channel.id, reason=reason, available_tags=list(new_tags.values())
+            ),
+        )
+        # TODO: update channel state here so that subsequent edits don't send old state?
+        for tag in channel_data.get("available_tags", []):
+            if int(tag["id"]) == self.id:
+                return ThreadTag(data=tag, channel=self._channel, state=self._state)
+        raise InvalidData("Could not find tag in response")
```

Comment:
Was this ever raised in your testing?

Reply:
nope, should only happen in case the api is broken somehow